src/components/signupform/SignupForm.js

```diff
@@ -1,110 +1,145 @@
-import { useState } from 'react'
-import CustomizableButton from '../common/CustomizableButton'
-import styles from './signupform.module.css'
+import { useState } from "react";
+import CustomizableButton from "../common/CustomizableButton";
+import styles from "./signupform.module.css";
+import { useAuthSignUp } from "@/composables/authSignUp";
 
 function Form() {
   const [user, setUser] = useState({
-    name: '',
-    email: '',
-    password: '',
-  })
+    name: "",
+    email: "",
+    password: "",
+  });
+
+  const { addUser, signup } = useAuthSignUp();
+  const [error, setError] = useState({});
+
   const handleChange = (e) => {
-    setUser({ ...user, [e.target.name]: e.target.value })
-  }
+    setUser({ ...user, [e.target.name]: e.target.value });
+  };
 
   const handleSubmit = (e) => {
-    e.preventDefault()
+    e.preventDefault();
+
     if (validateForm()) {
-      console.log('Form submitted sucessfully')
-      // Will direct the user to the Login Form
+      signup(user);
+
       setTimeout(() => {
-        resetForm()
-      }, 1000)
+        resetForm();
+      }, 1000);
     }
-  }
+  };
 
   const resetForm = () => {
-    setUser({ name: '', email: '', password: '' })
-  }
+    setUser({ name: "", email: "", password: "" });
+  };
 
+  // Validate Submit Form
   const validateForm = () => {
-    if (!user.name || !user.email || !user.password) {
-      alert('please fill out all required fields.')
-      return false
+    const errors = {};
+
+    // Validate Name
+    const namePattern = /^[a-zA-Z]{2,50}(?:\s+[a-zA-Z]{2,50})+$/;
```

Comment:
Create a validation module in a helper folder and move these validation-related functions there. Then import the functions from that module here and just call them with the necessary parameters.

Reply:
I've updated the file @jjleigh